libminifi/include/core/json/JsonNode.h

```diff
@@ -0,0 +1,248 @@
+    return node_ ? node_->IsObject() : false;
+  }
+  bool isNull() const override {
+    return node_ ? node_->IsNull() : false;
+  }
+
+  nonstd::expected<std::string, std::exception_ptr> getString() const override {
+    try {
+      if (!node_) {
+        throw std::runtime_error("Cannot get string of invalid json value");
+      }
+      if (!node_->IsString()) {
+        throw std::runtime_error("Cannot get string of non-string json value");
+      }
+      return std::string{node_->GetString(), node_->GetStringLength()};
+    } catch (...) {
+      return nonstd::make_unexpected(std::current_exception());
+    }
+  }
+
+  nonstd::expected<int, std::exception_ptr> getInt() const override {
+    return getNumber<int>("int");
+  }
+  nonstd::expected<unsigned int, std::exception_ptr> getUInt() const override {
+    return getNumber<unsigned int>("unsigned int");
+  }
+  nonstd::expected<int64_t, std::exception_ptr> getInt64() const override {
+    return getNumber<int64_t>("int64_t");
+  }
+  nonstd::expected<uint64_t, std::exception_ptr> getUInt64() const override {
+    return getNumber<uint64_t>("uint64_t");
+  }
+
+  nonstd::expected<bool, std::exception_ptr> getBool() const override {
+    try {
+      if (!node_) {
+        throw std::runtime_error("Cannot get bool of invalid json value");
+      }
+      if (!node_->IsBool()) {
+        throw std::runtime_error("Cannot get bool of non-bool json value");
+      }
+      return node_->GetBool();
+    } catch (...) {
+      return nonstd::make_unexpected(std::current_exception());
+    }
+  }
+
+  std::string getDebugString() const override {
+    if (!node_) return "<invalid>";
+    if (node_->IsObject()) return "<Map>";
+    if (node_->IsArray()) return "<Array>";
+    if (node_->IsNull()) return "null";
+    if (node_->IsInt()) return std::to_string(node_->GetInt());
+    if (node_->IsUint()) return std::to_string(node_->GetUint());
+    if (node_->IsInt64()) return std::to_string(node_->GetInt64());
+    if (node_->IsUint64()) return std::to_string(node_->GetUint64());
+    if (node_->IsTrue()) return "true";
+    if (node_->IsFalse()) return "false";
+    if (node_->IsDouble()) return std::to_string(node_->GetDouble());
+    if (node_->IsString()) return '"' + std::string(node_->GetString(), node_->GetStringLength()) + '"';
+    return "<unknown>";
+  }
+
+  size_t size() const override {
+    if (!node_) {
+      throw std::runtime_error("Cannot get size of invalid json value");
+    }
+    if (!node_->IsArray()) {
+      throw std::runtime_error("Cannot get size of non-array json value");
+    }
+    return node_->Size();
+  }
+  flow::Node::Iterator begin() const override;
+  flow::Node::Iterator end() const override;
+
+  flow::Node operator[](std::string_view key) const override {
+    if (!node_) {
+      throw std::runtime_error("Cannot get member of invalid json value");
+    }
+    if (!node_->IsObject()) {
+      return flow::Node{std::make_shared<JsonNode>(nullptr)};
```

Comment:
changed the implementation to be in-sync with the behavior of `YAML::Node::operator[]`

Reply:
Now it is the other branch which returns an invalid value instead of throwing (which I would expect).  I'm not sure we have to reproduce every weird behavior in `YAML::Node`, but if you think it's useful, I don't mind.

Reply:
the current parsing logic assumes the yaml behavior, I don't think it is useful and IMO we should throw as early as possible, but I didn't want to make larger changes to the logic, implementing a "stricter" behavior for json while maintaining yaml backwards compatibility should come in part2